pkg/config/event.go

```diff
@@ -0,0 +1,40 @@
+package config
+
+import (
+	"context"
+	"sync"
+)
+
+type Event[T any] struct {
+	mu          *sync.Mutex
+	subscribers []EventHandler[T]
+}
+
+func NewEvent[T any]() *Event[T] {
+	return &Event[T]{
+		subscribers: make([]EventHandler[T], 0),
+		mu:          new(sync.Mutex),
+	}
+}
```

Comment:
@shan-96 try to think of it as config service that provides certain functionality, here example how it could look like from architecture point of view, I'd try to isolate interaction with etcd on service side, became updates coming from etcd should be on driving side 
![image](https://github.com/user-attachments/assets/cd192fff-ab00-4305-98a6-7b0561f456db)

Reply:
yes, but there are two config sources (etcd and file) and we need to watch for changes on both. I can explain in a call maybe on weekend what I am trying to do

Reply:
it's fine to provide as an option to watch updates from file system, but not the same time when we watching updates from etcd, there should be only one source of truth

Reply:
correct - in future PRs I plan to add resolver before we update the config in the models.

watcher -> event -> resolver -> call repo

notice the `source` interface has a priority method to mitigate config merge collisions.

Reply:
for now you can assume to etcd config is not being used and we can look at how to integrate it later.